### PR TITLE
FRONT-1317: Migrate to a GraphQL client

### PR DESCRIPTION
**Pull request by mondoreale**
Quite a PR. I mixed few things into it on top of the "migration".

@pmichalowski check out `useSponsorship` and `useSponsorshipsList` hooks (our hooks) for how I query ze SUBgraphs.

---

A picture of a random SUBmarine to catch an eye of a larger audience:

image

### CODE REVIEW

**Comment by mondoreale** [comment]
I see the 🔴.

**Comment by mondoreale** [comment]
🟠 is the new 🟢, eh?

File: src/getters/du.ts [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+              }
+            : {}),
+    })
+
+    return new (await require('@dataunions/client')).DataUnionClient(
+        clientConfig,
+    ) as DataUnionClient
+}
+
+export async function getDataUnionSecrets(dataUnionId: string, chainId: number) {
```
Comment (pmichalowski) [inline src/getters/du.ts:100]: maybe we could add a return type here?

File: src/marketplace/containers/ProjectPage/useDataUnionServerStats.ts [hunk 2]
```diff
@@ -1,11 +1,15 @@
...
+
 const DATA_UNION_SERVER_POLL_INTERVAL_MS = 10000
 
 function useDataUnionServerStats() {
-    const [totalEarnings, setTotalEarnings] = useState(undefined)
-    const [memberCount, setMemberCount] = useState(undefined)
+    const [totalEarnings, setTotalEarnings] = useState<number>()
+
+    const [memberCount, setMemberCount] =
+        useState<Awaited<ReturnType<typeof getDataUnionStats>>['memberCount']>()
```
Comment (pmichalowski) [inline src/marketplace/containers/ProjectPage/useDataUnionServerStats.ts:11]: What's this "Awaited" type?

File: src/consts/index.tsx [hunk 3]
```diff
@@ -31,3 +31,5 @@ export function HubRouter({ history: historyProp = history, ...props }: HubRoute
         />
     )
 }
+
+export const MaxSearchPhraseLenght = 250
```
Comment (pmichalowski) [inline src/consts/index.tsx:35]: MaxSearchPhraseLeng**th** :)

**Review by pmichalowski — COMMENTED** [review]
we're missing a lot of return types in the `getters`, meaning:
- src/getters/hub.ts
- src/getters/du.ts
etc
but besides that it looks very good! nice job

File: src/marketplace/containers/ProjectPage/useDataUnionServerStats.ts (hunk 2 above)
Comment (mondoreale) [reply src/marketplace/containers/ProjectPage/useDataUnionServerStats.ts:11]: [`Awaited`](https://www.typescriptlang.org/docs/handbook/utility-types.html#awaitedtype)

File: src/consts/index.tsx (hunk 3 above)
Comment (mondoreale) [reply src/consts/index.tsx:35]: Good [cahct](https://www.google.com/search?q=catch)!

**Comment by mondoreale** [comment]
Thanks for throwing an eye, @pmichalowski! I'll address your feedback and get it merged by EOD.

AFK(ish) ATM. BBL.

**Pull request merged**